src/pages/SettingsPage.js

```diff
@@ -0,0 +1,172 @@
+  };
+  const editSyncPassword = newSyncPassword =>
+    setState({ ...state, modalKey: '', syncPassword: newSyncPassword });
+
+  const save = enteredPassword => {
+    const currentUserPassowrd = currentUser?.passwordHash ?? '';
+    const passwordMatch = hashPassword(enteredPassword) === currentUserPassowrd;
+    const toastMessage = passwordMatch
+      ? 'New details have been saved'
+      : 'Password did not match. New details have not been saved.';
+
+    if (passwordMatch) {
+      UIDatabase.write(() => {
+        UIDatabase.update('Setting', {
+          key: SETTINGS_KEYS.SYNC_URL,
+          value: syncURL,
+        });
+
+        UIDatabase.update('Setting', {
+          key: SETTINGS_KEYS.SYNC_SITE_PASSWORD_HASH,
+          value: hashPassword(syncPassword),
+        });
+      });
+    }
+
+    ToastAndroid.show(toastMessage, ToastAndroid.LONG);
+    closeModal();
+  };
+
+  const getModalSelect = () => {
+    switch (modalKey) {
+      case MODAL_KEYS.SYNC_URL_EDIT:
+        return editSyncURL;
+      case MODAL_KEYS.SYNC_PASSWORD_EDIT:
+        return editSyncPassword;
+      case MODAL_KEYS.CONFIRM_USER_PASSWORD:
+        return save;
+      default:
+        return null;
+    }
+  };
+
+  const { menuButton, menuButtonText: buttonText } = globalStyles;
+  const { row, column, marginRight, topRow } = styles;
+
+  const MenuButton = useCallback(
+    props => <Button style={menuButton} textStyle={buttonText} {...props} />,
+    []
+  );
+
+  const pageInfoColumns = useMemo(
+    () => [
+      [
+        {
+          title: 'Sync URL:',
+          editableType: 'text',
+          info: syncURL,
+          onPress: () => openModal(MODAL_KEYS.SYNC_URL_EDIT),
+        },
+        {
+          title: 'Sync password:',
+          editableType: 'text',
+          info: '',
+          onPress: () => openModal(MODAL_KEYS.SYNC_PASSWORD_EDIT),
+        },
+      ],
+    ],
+    [syncURL]
+  );
+
+  return (
+    <DataTablePageView>
+      <TouchableOpacity style={topRow} onPress={onSave}>
+        <Text style={marginRight}>Save Changes</Text>
+        <ConfirmIcon />
+      </TouchableOpacity>
+
+      <View style={row}>
+        <View style={column}>
+          <PageInfo columns={pageInfoColumns} />
+        </View>
+        <View>
+          <MenuButton text="Realm Explorer" onPress={toRealmExplorer} />
+          <MenuButton text="Export Data" onPress={exportData} />
+        </View>
+      </View>
+
+      <DataTablePageModal
+        fullScreen={false}
+        isOpen={!!modalKey}
+        modalKey={modalKey}
+        currentValue={modalKey === MODAL_KEYS.SYNC_URL_EDIT ? syncURL : ''}
+        onClose={closeModal}
+        onSelect={getModalSelect(modalKey)}
+      />
+    </DataTablePageView>
+  );
+};
+
+const mapStateToDispatch = dispatch => ({
+  toRealmExplorer: () => dispatch(gotoRealmExplorer()),
+});
+
+const mapStateToProps = state => ({
+  currentUser: state?.user?.currentUser,
```

Comment:
Just wondering it's always good to check nullish? If state is `null` then ahhhhh? currentUser `isRequired` in propTypes anyway.

Reply:
i mean this is just stupidly excessive state and user are guaranteed 049e7eea57ed57df7b2fae6fd4f3b967a82158aa